.github/workflows/build-and-publish-docker.yml

```diff
@@ -0,0 +1,48 @@
+name: Build Docker image
+
+on:
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}-build-image
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    # Buildx is currently required to use a subdirectory w/ build-push-action
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Log in to container registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
```

Comment:
This is currently using user permissions. I assume the secret is a personal access token? In the long run we don't want CI tied to individual users if possible.

Relevant section in the docs is 

https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#granular-permissions-for-userorganization-scoped-packages

but I think "organization account" refers to enterprise organization accounts. Happy to be wrong on that. 

A fallback would be to utilise a shared CI account.

Reply:
I've addressed this in the latest commit. The secret is a special github actions authentication token that isn't tied to a user: https://docs.github.com/en/actions/security-guides/automatic-token-authentication